internal/process/steps/gardener_cluster_test.go

```diff
@@ -119,6 +124,74 @@ spec:
 `, k8sClient)
 }
 
+func TestCheckGardenerCluster_RunWhenReady(t *testing.T) {
+	// given
+	os := storage.NewMemoryStorage().Operations()
+	existingGC := NewGardenerCluster("runtime-id-000", "kcp-system")
+	existingGC.SetState("Ready")
+	k8sClient := fake.NewClientBuilder().WithRuntimeObjects(existingGC.ToUnstructured()).Build()
+	svc := NewCheckGardenerCluster(os, k8sClient)
+	operation := fixture.FixProvisioningOperation("op", "instance-id")
+	operation.KymaResourceNamespace = "kcp-system"
+	operation.RuntimeID = "runtime-id-000"
+	operation.ShootName = "c-12345"
+	os.InsertOperation(operation)
+
+	// when
+	_, backoff, err := svc.Run(operation, logrus.New())
+
+	// then
+	assert.NoError(t, err)
+	assert.Zero(t, backoff)
+}
+
+func TestCheckGardenerCluster_RunWhenNotReady_OperationFail(t *testing.T) {
+	// given
+	os := storage.NewMemoryStorage().Operations()
+	existingGC := NewGardenerCluster("runtime-id-000", "kcp-system")
+	existingGC.SetState("In progress")
+	existingGC.obj.Object["status"] = map[string]interface{}{
+		"conditions": "some condition",
+	}
+	k8sClient := fake.NewClientBuilder().WithRuntimeObjects(existingGC.ToUnstructured()).Build()
+	svc := NewCheckGardenerCluster(os, k8sClient)
+	operation := fixture.FixProvisioningOperation("op", "instance-id")
+	operation.KymaResourceNamespace = "kcp-system"
+	operation.RuntimeID = "runtime-id-000"
+	operation.ShootName = "c-12345"
+	operation.UpdatedAt = time.Now().Add(-1 * time.Hour)
+	os.InsertOperation(operation)
+
+	// when
+	op, backoff, err := svc.Run(operation, logrus.New())
+	assert.Error(t, err)
+	assert.Zero(t, backoff)
+	assert.Equal(t, domain.Failed, op.State)
+}
+
+func TestCheckGardenerCluster_RunWhenNotReady_Retry(t *testing.T) {
+	// given
+	os := storage.NewMemoryStorage().Operations()
+	existingGC := NewGardenerCluster("runtime-id-000", "kcp-system")
```

Comment:
We could extract creating this fixture to avoid code duplication.

Reply:
I'm not sure which part should be extracted?

Reply:
I meant creating and setting GardenerCluster